packages/cli/src/hygraph/migrations/dynamicRow.ts

```diff
@@ -0,0 +1,242 @@
+      String: {
+        matches: {
+          flags: ['i', 's'],
+          regex: '^[a-z0-9-_.]+$',
+          errorMessage: 'Only letters, numbers, dashes (-), underscores (_) or dots allowed (.)',
+        },
+      },
+    },
+  })
+
+  client.createSimpleField({
+    displayName: 'Value',
+    apiId: 'value',
+    type: SimpleFieldType.String,
+    parentApiId: 'ConditionText',
+    isRequired: true,
+  })
+
+  client.createSimpleField({
+    displayName: 'Property',
+    apiId: 'property',
+    type: SimpleFieldType.String,
+    parentApiId: 'ConditionNumber',
+    isRequired: true,
+    validations: {
+      String: {
+        matches: {
+          flags: ['i', 's'],
+          regex: '^[a-z0-9-_.]+$',
+          errorMessage: 'Only letters, numbers, dashes (-), underscores (_) or dots allowed (.)',
+        },
+      },
+    },
+  })
+
+  client.createEnumerableField({
+    displayName: 'Operator',
+    apiId: 'operator',
+    parentApiId: 'ConditionNumber',
+    enumerationApiId: 'DynamicRowConditionNumberOperator',
+    isRequired: true,
+  })
+
+  client.createSimpleField({
+    displayName: 'Value',
+    apiId: 'value',
+    type: SimpleFieldType.Float,
+    parentApiId: 'ConditionNumber',
+    isRequired: true,
+  })
+
+  // ? MODEL
+  client.createModel({
+    apiId: 'DynamicRow',
+    apiIdPlural: 'DynamicRows',
+    displayName: 'Dynamic Row',
+    description:
+      'Dynamic rows allow you to add specific Row models to pages based on the properties of the page',
+  })
+
+  client.createSimpleField({
+    displayName: 'Internal name',
+    apiId: 'internalName',
+    description: 'Only used for internal reference',
+    type: SimpleFieldType.String,
+    isTitle: true,
+    isRequired: true,
+    isUnique: true,
+    modelApiId: 'DynamicRow',
+  })
+
+  client.createUnionField({
+    displayName: 'Row',
+    apiId: 'row',
+    reverseField: {
+      modelApiIds: ['RowQuote', 'RowLinks', 'RowColumnOne', 'RowColumnTwo', 'RowColumnThree'],
+      apiId: 'dynamicRow',
+      displayName: 'DynamicRows',
+      visibility: VisibilityTypes.Hidden,
+      isList: true,
+    },
+    parentApiId: 'DynamicRow',
+  })
+
+  client.createEnumerableField({
+    displayName: 'Placement',
+    apiId: 'placement',
+    parentApiId: 'DynamicRow',
+    enumerationApiId: 'DynamicRowPlacement',
+    description: 'Where will the row be placed relative to the target',
+    isRequired: true,
+  })
+
+  client.createUnionField({
+    displayName: 'Placement target',
+    apiId: 'target',
+    description:
+      'Optional: When the target is left blank it will place the Dynamic Row on the start or end.',
+    reverseField: {
+      modelApiIds: [
+        'RowQuote',
+        'RowLinks',
+        'RowColumnOne',
+        'RowColumnTwo',
+        'RowColumnThree',
+        'RowServiceOptions',
+        'RowContentLinks',
+        'RowButtonLinkList',
+        'RowProduct',
+        'RowSpecialBanner',
+        'RowHeroBanner',
+        'RowBlogContent',
```

Comment:
This will create a problem when adding new rows, if a new row is added it needs to be added here as well. Since Row*  models are expected to be developer-customisable, we need to properly sync these..